README.md

```diff
@@ -4,9 +4,34 @@ Iter-tools
 ![coverage functions](coverage/badge-functions.svg?sanitize=true)
 [![npm version](https://img.shields.io/npm/v/iter-tools.svg)](https://www.npmjs.com/package/iter-tools)
 
-iter-tools is an utility toolbox that allows you to unleash the power and expressiveness of iterables.
+iter-tools is designed to be a standard library of utilities for working with iterables. All javascript data types designed for the storage of data (namely Array, Map, and Set) are iterable, and iter-tools also includes some utilities that will help you work with objects. Working with iterables has powerful benefits:
+- The functional style helps you eliminate null pointer errors
+- It helps you use Maps and Sets, which otherwise have no tooling, yet offer attractive benefits like excellent performance, any-type keys, and the guarantee that prototype and data won't accidentally intermingle, causing difficult-to-find bugs.
+- It helps you create applications whose memory usage can be more highly optimized, helping you avoid costly garbage collections.
+- APIs which accept iterables are immediately compatible with almost any kind of data structure, including custom implementations such as those provided by Immutable.js
 
-If you want some ideas about how and when Iterables and iter-tools can help you out, take a look at [The Cookbook](https://github.com/sithmel/iter-tools/blob/master/COOKBOOK.md).
+If you want even more ideas about how and when Iterables and iter-tools can help you out, take a look at [The Cookbook](https://github.com/sithmel/iter-tools/blob/master/COOKBOOK.md).
+
+## Why iter-tools?
+
+Iter-tools is at present the only fully-featured library of its kind. Here is what it supports:
+
+- **Isomorphic**: The tools run equally well in a browser and in node.js.
+
+- **Currying/Partial Application**: iter-tools methods can be passed their arguments all at once, or can be incrementally configured. E.g. either `map(x => ++x)(iterable)` is valid. Omitting the iterable argument allows methods to be pipelined easily, as the output of one simple transform will be suitable as the only unbound argument for the next. Iter-tools provides `pipe` and `execPipe` functions for this kind of chaining.
+
+- **null/undefined are empty iterables**: iter-tools methods treat both null and undefined as empty iterables. This allows you to write transforms on array-ish data without first needing to check whether the array exists.
+
+- **Multiple iteration**: iter-tools transforms (as of v7.0.0) can be evluated multiple times if the source iterator can be. Not all iterables can be evaluated multiple times though: generator functions, notably, cannot be. Arrays, Maps, and Sets (and null/undefined) always can be.
```

Comment:
The way you implemented it should work fine with generator functions (I tested it with range)

Reply:
Are you sure, because that doesn't make any sense. Range is tricky since it can never be exhausted, but you can never get it to go back to the beginning either. Any generator function that can be exhausted will just be stuck at the end, and attempting to iterate over it again will just result in an empty iteration as its first yield will be `{done: true}`.

Reply:
I tested it again, and I was wrong.

Reply:
Hey, I tried reimplementing range as:
```js
export default function range (opts) {
  return {
    [Symbol.iterator]: function * _range () {
      let start, step, end
      opts = typeof opts === 'number' ? { end: opts, start: 0 }
        : (typeof opts === 'object' ? opts : {})
      step = opts.step === undefined ? 1 : opts.step
      end = opts.end === undefined
        ? (step > 0 ? Infinity : -Infinity) : opts.end
      start = opts.start ? opts.start : 0

      for (let i = start; step > 0 ? i < end : i > end; i += step) {
        yield i
      }
    }
  }
}
```
And now it can be replayed as many times you need.
Combinatory functions already return iterables, so they works already

Reply:
Yep, great change.